model/SyncLog/Storage/RdsSyncLogStorage.php

```diff
@@ -0,0 +1,253 @@
+             ->set(SyncLogStorageInterface::COLUMN_DATA, $qb->createNamedParameter(json_encode($entity->getData())))
+             ->set(SyncLogStorageInterface::COLUMN_REPORT, $qb->createNamedParameter(json_encode($entity->getReport())));
+
+         $finishTime = $entity->getFinishTime();
+         if ($finishTime instanceof \DateTime) {
+             $finishedAt = $entity->getFinishTime()->format(SyncLogEntity::DATE_TIME_FORMAT);
+             $qb->set(SyncLogStorageInterface::COLUMN_FINISHED_AT, $qb->createNamedParameter($finishedAt));
+         }
+
+         $qb->where(SyncLogStorageInterface::COLUMN_ID . ' = ' . $qb->createNamedParameter($entity->getId()));
+
+         return $qb->execute();
+     }
+
+     /**
+      * @param $id
+      * @return SyncLogEntity
+      * @throws common_exception_NotFound
+      * @throws InvalidServiceManagerException
+      */
+     public function getById($id)
+     {
+         $queryBuilder = $this->getQueryBuilder();
+         $queryBuilder->select('*')
+             ->where(SyncLogStorageInterface::COLUMN_ID . ' = ' . $queryBuilder->createNamedParameter($id, \PDO::PARAM_INT));
+
+         $data = $queryBuilder->execute()->fetchAll(\PDO::FETCH_ASSOC);
+
+         if (count($data) != 1) {
+             throw new common_exception_NotFound('There is no synchronization log record for provided ID.');
+         }
+
+         return $data[0];
+     }
+
+     /**
+      * @param $syncId
+      * @param $boxId
+      * @return SyncLogEntity
+      * @throws common_exception_NotFound
+      * @throws InvalidServiceManagerException
+      */
+     public function getBySyncIdAndBoxId($syncId, $boxId)
+     {
+         $queryBuilder = $this->getQueryBuilder();
+         $queryBuilder->select('*')
+             ->from(self::TABLE_NAME)
+             ->where(SyncLogStorageInterface::COLUMN_SYNC_ID . ' = ' . $queryBuilder->createNamedParameter($syncId, \PDO::PARAM_INT))
+             ->andWhere(SyncLogStorageInterface::COLUMN_BOX_ID . ' = ' . $queryBuilder->createNamedParameter($boxId, \PDO::PARAM_STR));
+
+         $data = $queryBuilder->execute()->fetchAll(\PDO::FETCH_ASSOC);
+         if (count($data) != 1) {
+             throw new common_exception_NotFound('There is no unique synchronization log record.');
+         }
+
+         return $data[0];
+     }
+
+     /**
+      * Get total amount of synchronization logs by provided filters.
+      *
+      * @param SyncLogFilter $filter
+      * @return integer
+      */
+     public function count(SyncLogFilter $filter)
+     {
+         try {
+             $qb = $this->getQueryBuilder()
+                 ->select('COUNT(*)')
+                 ->from(self::TABLE_NAME);
+
+             $this->applyFilters($qb, $filter);
+
+             return (int) $qb->execute()->fetchColumn();
+         } catch (\Exception $e) {
+             $this->logError('Counting synchronization logs failed: '. $e->getMessage());
+         }
+
+         return 0;
```

Comment:
Not very important but this can go in the catch part (as in the search() method)

Reply:
Fixed